server/game/GameActions/RandomDiscardAction.js

```diff
@@ -23,17 +23,17 @@ class RandomDiscardAction extends PlayerAction {
     }
 
     getEvent(player, context) {
-        return super.createEvent('unnamedEvent', { player, context }, () => {
-            let amount = Math.min(this.amount, player.hand.length);
-            if (this.location === 'archives') {
-                amount = Math.min(this.amount, player.archives.length);
-            }
+        let amount = Math.min(this.amount, player.hand.length);
+        if (this.location === 'archives') {
+            amount = Math.min(this.amount, player.archives.length);
+        }
 
-            let cards = _.shuffle(player.hand).slice(0, amount);
-            if (this.location === 'archives') {
-                cards = _.shuffle(player.archives).slice(0, amount);
-            }
+        let cards = _.shuffle(player.hand).slice(0, amount);
+        if (this.location === 'archives') {
+            cards = _.shuffle(player.archives).slice(0, amount);
+        }
 
```

Comment:
These changes are significant.  You're determining which cards (and how many cards) are discarded when the event is created, rather than when it resolves.  This could potentially cause problems (if a card is randomly selected to be discarded, but then is discarded by another effect in the interim), or it could be correct (if we ever got a card which said 'when this card would be discarded from your hand, do X', setting the event like this is probably how we would implement it).

For what you need here, I'm not sure this change is necessary: ```
return super.createEvent('unnamedEvent', { player, context }, event => {
    event.cards = cards```
achieves the same thing

Reply:
Fixed. Thanks!